### DISCUSSION TITLE
Upgrading tools installed via `rye install`

### DISCUSSION THREAD

**Discussion by dimfeld** [astral-sh/rye#496] [Q&A]
Is there a recommended way to upgrade all the tools installed by `rye install`?

The closest I've found so far is to iterate over the output of `rye tools list` and run `rye install -f $tool` on each one, but this has two issues:

- It unnecessarily reinstalls tools which are already at the latest version
- It doesn't retain any pinned python interpreter version for the tool (because Python is apparently the only language where a ton of packages refuse to install on the latest minor version)

**Comment by ischaojie** [comment]
In my opinion, it would be nice to provide commands like `rye tools update <some package>` or `rye tools update --all`.

**Comment by rizumu** [comment]
What about checking the cheeseshop to list outdated versions too?  That would be helpful info before deciding to upgrade all.

https://pip.pypa.io/en/stable/cli/pip_list/#cmdoption-o

**Comment by dimfeld** [comment]
If you've switched to `uv`, it supports this natively now, through `uv tool update --all`

**Reply by eli-yip** [reply]
Will this feature be back-ported to rye?
